Fixed #138 + Minor update to ReadMe

prithvi2k2 opened the pull request:
- Added `required` attribute to input field which now won't allow users to search with empty input and thus, won't reroute to 404 page. Fixes #138 

- Generalized Readme.md, it mentioned Hacktoberfest **2022** but the site is dynamic and works well in long-term, so I removed "**2022**".

max-programming commented:
Thanks! Can you also add a validation that if the search value is empty, it won't proceed

prithvi2k2 commented:
@max-programming my bad, I honestly thought `required` attribute will check for empty search values(like spaces only)... 
 updated code as follows:

- Will show the following error with `tailwind : animate-pulse` if an invalid input is entered - No empty input or no spaces only input - both aren't allowed. 
![image](https://github.com/max-programming/hacktoberfest-projects/assets/72189314/f19c593f-a7c1-4dec-8cb8-197683e73232)
- I validated it by trimming leading & trailing spaces of input and compared it with empty string. Hope this fixes the issue.
-  Also, removed the `required` attribute as `handleSubmit()` is checking for the same.

Please tell, if something still doesn't feel right.

prithvi2k2 commented:
@max-programming Is this PR worth the `hacktoberfest-accepted` label? I'll make more contributions soon...

max-programming commented:
@prithvi2k2 Don't worry about it getting accepted in hacktoberfest. Once it is merged, it will be approved

prithvi2k2 commented:
@max-programming Resolved a merge conflict with `Readme.md` file, hopefully, this can be merged now without a problem

prithvi2k2 commented on components/Hero.tsx:
```diff
@@ -10,11 +10,20 @@ import Button from './Button';
+    if (search === '') {
+      setErrorMessage('Empty search terms invalid!');
+      return;
+    } else {
```
Yeah, sure, I didn't refactor it after testing... my bad

prithvi2k2 replied:
Updated with commit [65e8759](https://github.com/max-programming/hacktoberfest-projects/pull/142/commits/65e8759413aa76e49b626ebc0292e399255c2c45)

max-programming commented:
Alright @prithvi2k2 
But the error message does not show up when I hit search with empty. Can you fix that?
Also, make sure to use one of the red colors in the tailwind config so that it matches the theme

max-programming commented:
And can you do the same for the searchbox on other pages?
![image](https://github.com/max-programming/hacktoberfest-projects/assets/51731966/4eb2a10e-1a40-43aa-bfc1-bcbfca2196a0)

prithvi2k2 commented:
Yes, the search other pages already doesn't take empty input... but for other inputs, I need to trim leading and trailing spaces to make it good... will try to do that

prithvi2k2 commented:
EXTRA: Found that search icon on home page when **hovered** is ugly due to high opacity hover bg color. Decreased the opacity to make it look good.

Before : 
![High Opacity BG](https://github.com/max-programming/hacktoberfest-projects/assets/72189314/46ddbe67-e3e4-4c6f-b21c-7052e758e884)


After: 
![Low Opacity BG](https://github.com/max-programming/hacktoberfest-projects/assets/72189314/df6054e0-43f5-459a-87d3-4b126952dec7)

prithvi2k2 commented:
> EXTRA: Found that search icon on home page when **hovered** is ugly due to high opacity hover bg color. Decreased the opacity to make it look good.
> 
> Before : ![High Opacity BG](https://user-images.githubusercontent.com/72189314/273240830-46ddbe67-e3e4-4c6f-b21c-7052e758e884.png)
> 
> After: ![Low Opacity BG](https://user-images.githubusercontent.com/72189314/273240934-df6054e0-43f5-459a-87d3-4b126952dec7.png)

Reverted the search icon change commit #c3a37d8 - as it's already in progress at #170 & #174 ... I didn't check mb. I shouldn't have been fixing a different issue in this PR, will not repeat that and try to create issue specific PRs only.

max-programming commented on components/Search.tsx:
```diff
@@ -40,7 +43,7 @@ export default function Search({ searchBarWrapperStyles }: SearchProps) {
             onClick={() => reset()}
           >
-            <GoX />
+            <GoX color='white'/>
```
Can you revert this because it's already fixed now?

max-programming replied:
I fixed it because there were conflicts

max-programming commented on components/Search.tsx:
```diff
@@ -19,7 +19,10 @@ export default function Search({ searchBarWrapperStyles }: SearchProps) {
   const onSubmit: SubmitHandler<FormValues> = ({ searchQuery }) => {
-    router.push({ query: { ...router.query, q: searchQuery } });
+    let trimmedQuery = searchQuery.trim();
+    if(trimmedQuery !== ''){ //Performs search only with non-empty strings
```
Just run prettier to format the code now

prithvi2k2 replied:
Added commit after formatting

prithvi2k2 commented:
Formatted with prettier.... And regarding the **clear input button** colour... please test it on light mode... I gave it white colour as it wasn't visible in light mode. You can redact it if it's unnecessary or tell me if I should revert that change... I didn't revert it now as it was marked resolved

max-programming commented:
> Formatted with prettier.... And regarding the **clear input button** colour... please test it on light mode... I gave it white colour as it wasn't visible in light mode. You can redact it if it's unnecessary or tell me if I should revert that change... I didn't revert it now as it was marked resolved

There were conflicts and I fixed it

max-programming commented:
@all-contributors please add @prithvi2k2 for code, design

max-programming commented:
@all-contributors please add @prithvi2k2 for code, design

prithvi2k2 commented:
Nevermind... let it be, continue with other work @max-programming np at all not being added in contributors list

max-programming commented:
@all-contributors please add @prithvi2k2 for code, design

max-programming commented:
> Nevermind... let it be, continue with other work @max-programming np at all not being added in contributors list

No worries. I fixed it ✌️
